lib/src/components/NxTabs/__tests__/NxStatefulTabs.test.tsx

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React from 'react';
+import { mount } from 'enzyme';
+import NxStatefulTabs from '../stateful/NxStatefulTabs';
+import NxTabList from '../../NxTabList/NxTabList';
+import NxTab from '../../NxTab/NxTab';
+import NxTabPanel from '../../NxTabPanel/NxTabPanel';
+
+describe('NxStatefulTabs', function () {
+  it('selects the first tab automatically', function () {
+    const component = mount(
```

Comment:
Could use `shallow` I'd think

Reply:
I'm testing that the `NxStatefulTabs` sets the appropriate `NxTab` as active by default which doesn't happen with `shallow`.

Reply:
> I'm testing that the NxStatefulTabs sets the appropriate NxTab as active by default which doesn't happen with shallow.

I would avoid re-testing the actual DOM behavior here that is implemented within `NxTabs`.  Instead I recommend only testing that `NxTabs` is receiving the expected props from the stateful wrapper at any given time.  And the of course `NxTabs` own tests should thoroughly test that those props get translated into the expected DOM.

Reply:
> When using a Context provider, I don't see how we can possibly use shallow to test that the correct context values are passed down to the child components since context isn't a prop of the child components. The only real way to confirm that the child components received the correct context is to check the behavior which requires mount.

None of the context stuff happens in the unit under test here.